src/components/body/Body.ts

```diff
@@ -180,8 +180,12 @@ export class DataTableBody implements OnInit, OnDestroy {
   }
 
   rowClicked(event, index, row): void {
-    this.onRowClick.emit({event, row});
-    this.selectRow(event, index, row);
+    let options = { cancel: false };
```

Comment:
Inside the rowClick event handler, I'm changing the options property of cancel to true if the condition is met.

I don't expect a race condition to occur.